src/models/User.js

```diff
@@ -1,32 +1,80 @@
-const { Schema, model } = require('mongoose');
+const {
+  Schema,
+  model,
+  Types: { ObjectId },
+} = require("mongoose");
+
 const emailRegex = /^[a-zA-Z0-9._%+-]{3,}@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
 // const emailRegex2 = /^[\w-]+(\.[\w-]+)*@[\w-]+(\.[\w-]+)+$/;
 
 //TODO: add validation for fields in the schema
 //TODO: add city and likedEvents
 const userSchema = new Schema({
-    username: { type: String, require: true, },
-    email: {
-        type: String, required: true, unique: true, validate: {
-            validator: (value) => emailRegex.test(value),
-            message: "Invalid email",
-
-        }
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+    validate: {
+      validator: (value) => emailRegex.test(value),
+      message: "Invalid email",
     },
-    firstName: { type: String, require: true },
-    lastName: { type: String, require: true },
-    isDeleted: {type: Boolean, default: false}
+  },
+  //TODO:  minimum and maximum length firstName
+  firstName: {
+    type: String,
+    require: true,
+    minlength: [2, "First name must be minimum 2 characters long!"],
+    maxlength: [15, "First name must be maximum 15 characters long!"],
+  },
+  //TODO:  minimum and maximum length lastName
+  lastName: {
+    type: String,
+    require: true,
+    minlength: [2, "Last name must be minimum 2 characters long!"],
+    maxlength: [15, "Last name must be maximum 15 characters long!"],
+  },
+  role: {
+    type: String,
+    require: true,
```

Comment:
Shouldn’t be required?

Reply:
Yes, my mistake! I will fix it in next commit!